R/zzz.R

```diff
@@ -71,8 +88,13 @@ get_widget <- function(comparison_item, category, gprop, hl) {
   
   url <- encode_keyword(url)
   
-  widget <- curl::curl_fetch_memory(url)
-
+  # if cookie_handler hasn't been set up, get the requisite cookies from Google's API
+  if(!exists("cookie_handler", envir = gtrendsR:::.__NAMESPACE__.)){ get_api_cookies() }
+  # if(!exists("cookie_handler", envir = as.environment('package:gtrendsR'))){ get_api_cookies() }
+  # if(!exists("cookie_handler")){ get_api_cookies() }
+  
+  widget <- curl::curl_fetch_memory(url, handle = gtrendsR:::.__NAMESPACE__.$cookie_handler) # I'm not sure you're supposed to do this but it works
```

Comment:
@eddelbuettel is very good with this Environment thing. I remember him doing it when it was necessary to log in. 

Could we just do:
```r
cookie_handler <- curl::new_handle()
  curl::curl_fetch_memory("http://apis.google.com/Cookies/OTZ", handle = cookie_handler)
  curl::handle_cookies(cookie_handler)
  
 widget <- curl::curl_fetch_memory(url, handle = cookie_handler)
```

Reply:
What is the problem? Persisting a cookie across the session?  `new.env()` and assigning is your friend.  [Here](https://github.com/eddelbuettel/rpushbullet/blob/b566f6da3a5774c88796d3122ba96961cf982add/R/init.R#L21-L38) is a good use case from `RPushbullet`, also see the later access to `.pkgenv` in the same file.  It's an "environment" so works as a hash, you could store and retrieve `.pkgenv[["cookie"]]` methinks.